Emby.Server.Implementations/LiveTv/EmbyTV/EncodedRecorder.cs

```diff
@@ -68,7 +68,8 @@ public async Task Record(IDirectStreamProvider directStreamProvider, MediaSource
             _logger.LogInformation("Recording completed to file {0}", targetFile);
         }
 
-        private async Task RecordFromFile(MediaSourceInfo mediaSource, string inputFile, string targetFile, Action onStarted, CancellationToken cancellationToken)
+        // Important - don't make this async or we won't be able to kill ffmpeg when the user stops playback
```

Comment:
I don't think that is the issue. I'm pretty sure you can make it async, but you need to add `await _taskCompletionSource.Task.ConfigureAwait(false);`

Reply:
Thanks for pointing that out, I somehow missed that `_taskCompletionSource` has state and didn't just provide a `Task` to return. Since it was removed when the `await` was added I fixated on the `await` and missed the important difference in that commit.